EIPS/eip-4671.md

```diff
@@ -0,0 +1,468 @@
+    /// @param operators Addresses allowed to mint a token
+    /// @param owners Addresses for whom `operators` are allowed to mint a token
+    function delegateBatch(address[] memory operators, address[] memory owners) public virtual override {
+        require(operators.length == owners.length, "operators and owners must have the same length");
+        bool isCreator = _isCreator();
+        for (uint i=0; i<operators.length; i++) {
+            _delegateAsDelegateOrCreator(operators[i], owners[i], isCreator);
+        }
+    }
+
+    /// @notice Mint a token. Caller must have the right to mint for the owner.
+    /// @param owner Address for whom the token is minted
+    function mint(address owner) public virtual override {
+        _mintAsDelegateOrCreator(owner, _isCreator());
+    }
+
+    /// @notice Mint tokens to multiple addresses. Caller must have the right to mint for all owners.
+    /// @param owners Addresses for whom the tokens are minted
+    function mintBatch(address[] memory owners) public virtual override {
+        bool isCreator = _isCreator();
+        for (uint i=0 ; i<owners.length; i++) {
+            _mintAsDelegateOrCreator(owners[i], isCreator);
+        }
+    }
+
+    /// @notice Check if an operator is a delegate for a given address
+    /// @param operator Address of the operator
+    /// @param owner Address of the token's owner
+    /// @return True if the `operator` is a delegate for `owner`, false otherwise
+    function isDelegate(address operator, address owner) public view returns (bool) {
+        return _allowed[operator][owner];
+    }
+
+    /// @notice Check if you are a delegate for a given address
+    /// @param owner Address of the token's owner
+    /// @return True if the caller is a delegate for `owner`, false otherwise
+    function isDelegateOf(address owner) public view returns (bool) {
+        return isDelegate(msg.sender, owner);
+    }
+
+    function supportsInterface(bytes4 interfaceId) public view virtual override(NTT) returns (bool) {
+        return 
+            interfaceId == type(INTTDelegate).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
+
+    function _delegateAsDelegateOrCreator(address operator, address owner, bool isCreator) private {
+        require(
+            isCreator || _allowed[msg.sender][owner],
+            "Only contract creator or allowed operator can delegate"
+        );
+        if (!isCreator) {
+            _allowed[msg.sender][owner] = false;
+        }
+        _allowed[operator][owner] = true;
+    }
+
+    function _mintAsDelegateOrCreator(address owner, bool isCreator) private {
+        require(
+            isCreator || _allowed[msg.sender][owner],
+            "Only contract creator or allowed operator can mint"
+        );
+        if (!isCreator) {
+            _allowed[msg.sender][owner] = false;
+        }
+        _mint(owner);
+    }
+}
+```
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+## NTT for EIP ?
```

Comment:
While the idea is novel, this also isn't an EIP-1 section, and should be removed. You could release it as a blog post or similar.